src/movies.js

```diff
@@ -1,22 +1,55 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+  const mappedArray = movies.map(function (singleMovie){
+    return singleMovie.director;
+   })
+   return mappedArray;
+}
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies){
+  const movieDirector = movies.filter( (directorName) => {
+    return directorName.director === 'Steven Spielberg';
+  })
+  for (let i=0; movieDirector === 'Steven Spielberg'; i++){
+    const directorDrama = movieDirector.filter( (dramaMovie) => {
+      return dramaMovie.genre === 'Drama';
+    })
+  }
+  return directorDrama;
 
+}
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+  let movieScores = movies.reduce(function (accValue, currentValue){
+    return accValue.score + currentValue.score
+  })
+  return movieScores
 
+}
 // Iteration 4: Drama movies - Get the average of Drama Movies
 function dramaMoviesScore() {}
```

Comment:
Steps for iteration 4:
1. create a function that receives array of movies
2. filter that array so we have only drama movies. --> do nto forget genre is an array, so you might want to use .include() method
3. use reduce method to add all movie scores
4. calculate and return the average score of drama movies